src/caffe/solver.cpp

```diff
@@ -469,6 +495,32 @@ void SGDSolver<Dtype>::ApplyUpdate() {
 }
 
 template <typename Dtype>
+void SGDSolver<Dtype>::Normalize(int param_id) {
+  if (this->param_.iter_size() == 1) { return; }
+  // Scale gradient to counterbalance accumulation.
+  const vector<shared_ptr<Blob<Dtype> > >& net_params = this->net_->params();
+  const Dtype accum_normalization = Dtype(1.) / this->param_.iter_size();
```

Comment:
Is this normalization correct?

Doing this will reduce the gradient by a factor of `iter_size` compared to computing the gradient over an entire batch. If I'm interpreting this correctly, learning rates should be multiplied by `iter_size` to overcome this existing code.

Or: Is learning rate automatically scaled by the batch size elsewhere, and this code is necessary to account for the effective increase in the batch size?